### DISCUSSION TITLE
AuthenticationException: Cannot determine the frame size or a corrupted frame was received

### DISCUSSION THREAD

**Discussion by kmai-universe** [rabbitmq/rabbitmq-dotnet-client#1401] [Q&A] [answered]
Hi,

For testing purposes i have spinned up a docker RabbitMQ server with default settings.

Then i have created a console test project with .NET 7

```
  var factory = new ConnectionFactory();
  
  factory.Uri = new Uri("amqps://test.user:test123@localhost:5672/");
  
  factory.ClientProvidedName = "test";
  
  var conn = factory.CreateConnection();
  var channel = conn.CreateModel();
  
  string queueName = "test.queue";
  string exchangeName = "test.fanout";
  
  channel.ExchangeDeclarePassive(exchangeName);
  
  channel.QueueDeclare(queueName);
  channel.QueueBind(queueName, exchangeName, "", null);
  
  var consumer = new EventingBasicConsumer(channel);
  
  consumer.Received += Consumer_Received;
  
  channel.BasicConsume(queue: queueName,
           autoAck: true,
           consumer: consumer);
  
  void Consumer_Received(object? sender, BasicDeliverEventArgs e)
  {
      var body = e.Body.ToArray();
      var message = Encoding.UTF8.GetString(body);
  
      Console.WriteLine(message);
  }
```

However the .createConnection call fails with an exception:

"None of the specified endpoints were reachable" and an inner exception:

"Cannot determine the frame size or a corrupted frame was received."

If i look in RabbitMQ i get the following error:

```
[info] <0.7264.0> accepting AMQP connection <0.7264.0> (172.17.0.1:43726 -> 172.17.0.2:5672)
[error] <0.7264.0> closing AMQP connection <0.7264.0> (172.17.0.1:43726 -> 172.17.0.2:5672):
[error] <0.7264.0> {bad_header,<<22,3,1,1,169,1,0,1>>}
```

Sorry for the lack of code formatting for the example project but i simply can't get the code block to work with the example!

**Comment by lukebakken** [comment]
Hello, please let us know exactly how you started your Docker container and created your test user.

Also, your program will exit right after the call to `channel.BasicConsume` ... is that what you intended? There is nothing keeping it running.

**Reply by kmai-universe** [reply]
I'm running Docker Desktop on my Windows 10 machine. I just searched for rabbitmq and pulled "rabbitmq:3.12.6-management-alpine" after which i mapped all the ports to the internal ports.

I created the exchange directly in the Rabbit web admin as a fanout exchange and i'm only running the default / virtual host. The user has full access to the virtual host / and the exchange.

Regarding the program exit, you're right, i forgot a key catch to keep the app running!

**Reply by kmai-universe** [reply]
Strangely if i connect through this method instead it works!

```
var factory = new ConnectionFactory() { HostName = "localhost", UserName = "test.user", Password = "test123" };
```

**Comment by lukebakken** [comment, marked as answer]
I see the issue, you are using `amqps` (i.e. TLS-encrypted AMQP) in your original Uri:

```
factory.Uri = new Uri("amqps://test.user:test123@localhost:5672/");
```

Replace `amqps` with `amqp` and re-try.
